doc/map-format.md

```diff
@@ -0,0 +1,188 @@
+# Map Format
+
+**What is this?** This document descrcibes the file format for save files of openage.
```

Comment:
So you propose a scenario file is exactly the same as a saved game state? Could be done, like a "snapshot".
Much of the gamestate is lost in this simple format, we need a full recording format as well.

I was thinking of using the ideas from https://blog.forrestthewoods.com/the-tech-of-planetary-annihilation-chronocam-292e3d6b169a :)

But still, this is waaay better than our current save format.